File: Net/POO/Empleado.cs
using System;

public class Empleado
{
    public string Nombre { get; set; }
    public string Apellido { get; set; }
    public decimal Salario { get; set; }

    public Empleado(string nombre, string apellido, decimal salario)
    {
        Nombre = nombre;
        Apellido = apellido;
        Salario = salario;
    }

    public decimal CalcularSalarioAnual()
    {
        return Salario * 12;
    }
}

public class Gerente : Empleado
{
    public string Departamento { get; set; }

    public Gerente(string nombre, string apellido, decimal salario, string departamento)
        : base(nombre, apellido, salario)
    {
        Departamento = departamento;
    }

    public void IncrementarSalario(decimal porcentaje)
    {
        Salario += Salario * (porcentaje / 100);
    }
}

public class Program
{
    public static void Main()
    {
        // Crear una instancia de la clase Gerente
        Gerente gerente = new Gerente("Ana", "García", 5000, "Ventas");

        // Incrementar el salario del gerente en un 10%
        gerente.IncrementarSalario(10);

        // Mostrar el nuevo salario anual
        Console.WriteLine($"El nuevo salario anual de {gerente.Nombre} {gerente.Apellido} es: {gerente.CalcularSalarioAnual()}");

        // Espera antes de cerrar
        Console.WriteLine("Presiona cualquier tecla para salir...");
        Console.ReadKey();
    }
}

File: Python/POO/Empleado.py
class Empleado:
    def __init__(self, nombre, apellido, salario):
        self.nombre = nombre
        self.apellido = apellido
        self.salario = salario

    def calcular_salario_anual(self):
        return self.salario * 12

    # Método getter para el salario
    def get_salario(self):
        return self.salario

    # Método setter para el salario
    def set_salario(self, nuevo_salario):
        if nuevo_salario > 0:
            self.salario = nuevo_salario
        else:
            print("El salario debe ser un valor positivo.")

# Crear una instancia de la clase Empleado
empleado = Empleado("Juan", "Pérez", 3000)

# Mostrar el salario anual
print(f"El salario anual de {empleado.nombre} {empleado.apellido} es: {empleado.calcular_salario_anual()}")

# Modificar el salario del empleado
empleado.set_salario(3500)

# Mostrar el nuevo salario anual
print(f"El nuevo salario anual de {empleado.nombre} {empleado.apellido} es: {empleado.calcular_salario_anual()}")


